materialize-sql/driver.go

```diff
@@ -287,16 +296,19 @@ func (d *Driver) newTransactor(ctx context.Context, open pm.TransactionRequest_O
 	var loadedVersion string
 
 	var endpoint, err = d.NewEndpoint(ctx, open.Materialization.EndpointSpecJson)
```

Comment:
I think we'd still want to unconditionally check for `err` here and return if it isn't `nil`, to guard against `endpoint` being `nil` and causing problems later